src/lib/tecs/TECS.hpp

```diff
@@ -49,351 +50,685 @@
+	 *
+	 * @param param is the control parameters.
+	 * @param flag is the control flags.
+	 * @return Weights used for the specific energy balance.
+	 */
+	SpecificEnergyWeighting _updateSpeedAltitudeWeights(const Param &param, const Flag &flag);
+	/**
+	 * @brief Calculate pitch control.
+	 *
+	 * @param dt is the update time intervall in [s].
+	 * @param input is the current input measurement of the UAS.
+	 * @param se is the calculated specific energy.
+	 * @param param is the control parameters.
+	 * @param flag is the control flags.
+	 */
+	void _calcPitchControl(float dt, const Input &input, const SpecificEnergyRates &se, const Param &param,
+			       const Flag &flag);
 
-	void set_speed_comp_filter_omega(float omega) { _tas_estimate_freq = omega; }
-	void set_speed_weight(float weight) { _pitch_speed_weight = weight; }
-	void set_airspeed_error_time_constant(float time_const) { _airspeed_error_gain = 1.0f / math::max(time_const, 0.1f); }
+	/**
+	 * @brief Calculate pitch control specific energy balance rates.
+	 *
+	 * @param weight is the weighting use of the potential and kinetic energy.
+	 * @param se is the specific energy rates
+	 * @return specific energy balance rates
+	 */
+	EnergyRateValues _calcPitchControlSeb(const SpecificEnergyWeighting &weight, const SpecificEnergyRates &se) const;
 
-	void set_throttle_damp(float throttle_damp) { _throttle_damping_gain = throttle_damp; }
-	void set_throttle_slewrate(float slewrate) { _throttle_slewrate = slewrate; }
+	/**
+	 * @brief Calculate the pitch control update function.
+	 * Update the states of the pitch control
+	 *
+	 * @param dt is the update time intervall in [s].
+	 * @param seb is the specific energy balance rate.
+	 * @param param is the control parameters.
+	 */
+	void _calcPitchControlUpdate(float dt, const EnergyRateValues &seb, const Param &param);
 
-	void set_roll_throttle_compensation(float compensation) { _load_factor_correction = compensation; }
-	void set_load_factor(float load_factor) { _load_factor = load_factor; }
+	/**
+	 * @brief Calculate the pitch control output function.
+	 *
+	 * @param input is the current input measurement of the UAS.
+	 * @param seb is the specific energy balance rate.
+	 * @param param is the control parameters.
+	 * @param flag is the control flags.
+	 * @return pitch setpoint angle [rad].
+	 */
+	float _calcPitchControlOutput(const Input &input, const EnergyRateValues &seb, const Param &param,
+				      const Flag &flag) const;
 
-	void set_ste_rate_time_const(float time_const) { _STE_rate_time_const = time_const; }
-	void set_speed_derivative_time_constant(float time_const) { _speed_derivative_time_const = time_const; }
+	/**
+	 * @brief Update controlled throttle setpoint.
+	 *
+	 * @param dt is the update time intervall in [s].
+	 * @param se is the calculated specific energy.
+	 * @param flag is the control flags.
+	 */
+	void _calcThrottleControl(float dt, const SpecificEnergyRates &se, const Param &param, const Flag &flag);
```

Comment:
I don't like this short abbreviation like `se` here, followed by `spe` and `ske`. but also see that writing it out is probably not helping much for readability. There's also the "rate" keyword missing. Maybe we could find a compromise of calling se = specific_energy_rates and ske = ske_rate, spe = spe_rate? And then remove the "rate" keyword from rate_stepoint and rate_error.